src/main/java/com/worksoft/jenkinsci/plugins/ctm/model/CTMServer.java

```diff
@@ -13,528 +13,613 @@
+     StringBuilder requestBody = new StringBuilder();
+     String requestUrl = this.ctmUrl + "api/SuiteExecution/" + suiteId + "/runtimeparameters";
+ 
+     requestBody = requestBody.append("{\"SuiteId\":\"").append(suiteId).append("\",\"SuiteName\":\"").append(suiteName)
+         .append("\",\"TenantId\":\"").append(tenantId).append("\",\"TenantName\":\"").append(tenantName)
+         .append("\",\"Processes\":[");
+ 
+     if (processIdCntr <= 0) {
+       for (CTMProcess ctm : processMap.values()) {
+         requestBody = requestBody.append("\",\"ProcessPath\":\"").append(ctm.GetProcessPath())
+             .append("\",\"Layout\":\"").append(ctm.GetLayout()).append("\",\"Recordset\":\"").append(ctm.GetRecordset())
+             .append("\",\"RecordsetMode\":\"").append(ctm.GetRecordsetMode()).append("\",\"MachineAttributes\":\"")
+             .append(ctm.GetMachineAttributes().replace(':', '=')).append("\"},");
+       }
+     } else {
+       for (CTMProcess ctm : processMap.values()) {
+         requestBody = requestBody.append("{\"ProcessId\":\"").append(ctm.GetProcessId()).append("\",\"ProcessPath\":\"")
+             .append(ctm.GetProcessPath())
+             .append("\",\"Layout\":\"").append(ctm.GetLayout()).append("\",\"Recordset\":\"").append(ctm.GetRecordset())
+             .append("\",\"RecordsetMode\":\"").append(ctm.GetRecordsetMode()).append("\",\"MachineAttributes\":\"")
+             .append(ctm.GetMachineAttributes().replace(':', '=')).append("\"},");
+       }
+     }
+     requestBody = requestBody.deleteCharAt(requestBody.lastIndexOf(","));
+     requestBody.append("]}");
+ 
+     details.consoleOut.println("Request body --->" + requestBody);
+     details.consoleOut.println("Get suite url: " + requestUrl);
+ 
+     HttpRequest execParamsRequest = HttpRequest
+         .post(this.ctmUrl + "api/SuiteExecution/" + suiteId + "/runtimeparameters");
+     execParamsRequest.contentType("application/json");
+     execParamsRequest.body(requestBody.toString());
+ 
+     String guid = null;
+ 
+     CTMResult result = sendRequest(execParamsRequest);
+     details.consoleOut.println("\n------execute suite-----------\n" + result.dumpDebug());
+     if (result.is200()) {
+       JSONObject json = result.getJsonData();
+       if (json.containsKey("SuiteExecutionResultId")) {
+         guid = json.getString("SuiteExecutionResultId");
+       }
+     } else {
+       String response = result.getResponseData();
+       if (response != null
+           && StringUtils.isNotEmpty(response)) {
+         details.consoleOut.println(result.dumpDebug());
+         if (response.contains("No machines are available for execution")) {
+           throw new RuntimeException("No machines are available for execution");
+         } else if (response.contains("No machine credentials are available for execution.")) {
+           throw new RuntimeException("No machine credentials are available for execution.");
+         } else {
```

Comment:
at first the if conditions for checking response.contains looked the same.    This is fine what you are doing.   However, since the last else is just throwing an exception with response, it might be easier to just have a consistent flow where you don't need these if statements and just always throw a RuntimeException with the response.   It doesn't look like there is an advantage for trying to inspect what is in the response.  Just my opinion.

Reply:
I agree, if the response message already has the error message text needed there is no need for the multiple checks.  However, if the response has extra text that we don't won't displayed in the error message then this approach is acceptable.

Also since all the if/else branches are throwing the same exception just with different messages the if/else blocks should just set the value of the message and then have one throw statement at the end.
Something like this would be preferred:
```
      if (StringUtils.isNotEmpty(response)) {
        details.consoleOut.println(result.dumpDebug());
        String errorMsg;
        if (response.contains("No machines are available for execution")) {
          errorMsg = "No machines are available for execution";
        } else if (response.contains("No machine credentials are available for execution.")) {
          errorMsg = "No machine credentials are available for execution";
        } else {
          errorMsg = "Execute suite failure - " + response;
        }
        if (StringUtils.isNotEmpty(errorMsg)) {
          throw new RuntimeException(errorMsg);
        }
      }
```

Reply:
Code has been modified as per above suggestions